wdl/src/bin/wdl.rs

```diff
@@ -103,7 +105,15 @@ async fn analyze<T: AsRef<dyn Rule>>(
         },
     );
 
-    analyzer.add_documents(vec![path]).await?;
+    // Convert the file to a either a local URL or a remote URL
+    let file = if file.starts_with("http://") || file.starts_with("https://") {
```

Comment:
Is this sufficient? What about other protocols such as FTP? Or URLs that omit the leading protocol?

Reply:
Would it be better to just always call `Url::parse()`, and if that fails attempt `PathBuf::from()` (and if that fails, `bail!()`)?

Reply:
Just typed up the following and it seems to work:
```rust
    let file = if let Ok(url) = Url::parse(&file) {
        url
    } else if let Some(url) = path_to_uri(&PathBuf::from(file.clone())) {
        url
    } else {
        bail!("failed to convert `{file}` to a URI", file = file)
    };
```

Reply:
That's what I was wondering. I'm curious to hear other thoughts on this, but it makes sense to me.

Reply:
I think it makes sense to attempt to parse it as a URI and fallback to treating it as a path (also, `path_to_uri` should probably be changed to take `impl AsRef<Path>` instead of `&Path`).

Regarding the supported schemes for the URIs, I think only `http`, `https`, and `file` should be supported as it's the only schemes required by the standard for import statements (with `file` currently deprecated and `http` should definitely be deprecated, but isn't currently).